### PR TITLE
Update class diagram to v3 using new renderer

### PR DESCRIPTION
**Pull request by yari-dewalt**
## :bookmark_tabs: Summary

Updates the class diagram to the new unified way of rendering. Includes a new "classBox" shape to be used in diagrams as well as some other updates such as: the option to hide the empty members box in class diagrams, support for handDrawn look, the introduction of the classDef statement into class diagrams, support for styling the default class, and lollipop interfaces.

Includes fixes / additions for #5562 #3139 and #4037

## :straight_ruler: Design Decisions

- The classDb gathers the necessary data and passes it to the v3 unified renderer.
- The classBox shape is kind of complicated due to the placement and possibility of different text (annotations and members as opposed to just a plain label) as well as taking in classNode information, so it's made a little awkward. It utilizes some helper functions defined in shapeUtil to accomplish this, although I am not sure this is the best way to go about it.
- The syntax of the class diagram and its use of specific symbols also conflicts with text creation in terms of how HTML and markdown is treated (opening and closing tags <>, - being used for lists, * being used at the end for a classifier, etc.) and so there are workarounds / hacks in place. I tried to not stray away from the typical process and so I am unsure about how acceptable these workarounds are and if they are the best method, but can revise and fix if needed.
- The classDef statement works similarly as it does in other diagrams and defines a class that can be used in the diagram (either via the shorthand ::: or a cssClass statement).
-  Support for lollipop interfaces using ()-- and --(). This was already a supported relationship between classes, due to #1145 but now properly creates a lollipop interface instead. I left ()--() in as of right now which creates a lollipop "relation" between two classes, but it is not an official UML class relationship so unsure about leaving this in.

### :clipboard: Tasks

Make sure you

- [x] :book: have read the [contribution guidelines](https://mermaid.js.org/community/contributing.html)
- [x] :computer: have added necessary unit/e2e tests.
- [x] :notebook: have added documentation. Make sure [`MERMAID_RELEASE_VERSION`](https://mermaid.js.org/community/contributing.html#update-documentation) is used for all new features.
- [ ] :butterfly: If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

**Review by jgreywolf — COMMENTED** [review]
Small comments from initial scan.  Have not run through code in vscode for more feedback yet :)

**Comment by jgreywolf** [comment]
@yari-dewalt What is the status of this?

File: packages/mermaid/src/config.type.ts [hunk 1]
```diff
@@ -716,6 +716,7 @@ export interface ClassDiagramConfig extends BaseDiagramConfig {
    */
   diagramPadding?: number;
   htmlLabels?: boolean;
+  hideEmptyMembersBox?: boolean;
```
Comment (knsv) [inline packages/mermaid/src/config.type.ts:719]: Good to configure

File: packages/mermaid/src/diagrams/class/classTypes.ts [hunk 2]
```diff
@@ -85,7 +90,7 @@ export class ClassMember {
           this.visibility = detectedVisibility as Visibility;
         }
 
-        this.id = match[2].trim();
+        this.id = match[2];
```
Comment (knsv) [inline packages/mermaid/src/diagrams/class/classTypes.ts:93]: Great to remove redundant code. Are you sure we never need the trim here?

File: packages/mermaid/src/diagrams/class/parser/classDiagram.jison [hunk 3]
```diff
@@ -326,6 +328,15 @@ noteStatement
     | NOTE noteText                { yy.addNote($2); }
     ;
 
+classDefStatement
```
Comment (knsv) [inline packages/mermaid/src/diagrams/class/parser/classDiagram.jison:331]: Looks good!

**Review by knsv — CHANGES_REQUESTED** [review]
Great job. I like the changes overall. Solid professional effort. Some issues found, the rendering does not work properly for cardinality when using elk.

**Comment by yari-dewalt** [comment]
@jgreywolf 
> @yari-dewalt What is the status of this?

Changes are fully reviewed now, so going through and fixing and then it should be good!

**Comment by yari-dewalt** [comment]
E2E tests failing because of screenshot capacity

> An error was thrown in your plugins file while executing the handler for the after:run event.
> 
> The error we received was:
> 
> Error: You have reached the maximum screenshot capacity included in your Pro open source xl Plan. Please upgrade your Plan.
>     at throwAPIError (file:///__w/mermaid/mermaid/node_modules/.pnpm/@argos-ci+api-client@0.5.0/node_modules/@argos-ci/api-client/dist/index.mjs:28:15)
>     at upload (file:///__w/mermaid/mermaid/node_modules/.pnpm/@argos-ci+core@2.8.1/node_modules/@argos-ci/core/dist/index.mjs:886:9)
>     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
>     at async Object.<anonymous> (file:///__w/mermaid/mermaid/node_modules/.pnpm/@argos-ci+cypress@2.2.2_cypress@13.15.0/node_modules/@argos-ci/cypress/dist/task.mjs:32:21)
>     at async Object.handlerThatCallsUserDefinedHandler [as handler] (/__w/mermaid/mermaid/node_modules/.pnpm/@applitools+eyes-cypress@3.44.9_encoding@0.1.13_typescript@5.4.5/node_modules/@applitools/eyes-cypress/dist/plugin/pluginExport.js:105:21)
> Test run failed, code 1
> More information might be available above
> Cypress module has returned the following error message:
> Could not find Cypress test run results
> Error: Could not find Cypress test run results

**Review by knsv — APPROVED** [review]

**Pull request merged**